editor/doc_tools.cpp

```diff
@@ -76,241 +76,275 @@ static String _translate_doc_string(const String &p_text) {
+				// so we have to check one by one so we make sure we have an exact match.
+				for (int64_t arg_i = 0; arg_i < arg_count; ++arg_i) {
+					for (int64_t arg_j = 0; arg_j < arg_count; ++arg_j) {
+						if (from.arguments[arg_i].type == to.arguments[arg_j].type && !arg_used[arg_j]) {
+							arg_used.write[arg_j] = true;
+							break;
 						}
 					}
-					if (not_the_same) {
-						continue;
+				}
+				bool not_the_same = false;
+				for (int64_t arg_i = 0; arg_i < arg_count; ++arg_i) {
+					if (!arg_used[arg_i]) { // At least one of the arguments was different.
+						not_the_same = true;
+						break;
 					}
 				}
-
-				const DocData::MethodDoc &mf = cf.constructors[j];
-
-				m.description = mf.description;
-				m.is_deprecated = mf.is_deprecated;
-				m.deprecated_message = mf.deprecated_message;
-				m.is_experimental = mf.is_experimental;
-				m.experimental_message = mf.experimental_message;
-				break;
+				if (not_the_same) {
+					continue;
+				}
 			}
+
+			to.description = from.description;
+			to.is_deprecated = from.is_deprecated;
+			to.deprecated_message = from.deprecated_message;
+			to.is_experimental = from.is_experimental;
+			to.experimental_message = from.experimental_message;
+			break;
 		}
+	}
+}
 
-		for (int i = 0; i < c.methods.size(); i++) {
-			DocData::MethodDoc &m = c.methods.write[i];
+static void merge_methods(Vector<DocData::MethodDoc> &p_to, const Vector<DocData::MethodDoc> &p_from) {
+	// Get data from `p_to`, to avoid mutation checks. Searching will be done in the sorted `p_to` from the (potentially) unsorted `p_from`.
+	DocData::MethodDoc *to_ptrw = p_to.ptrw();
+	int64_t to_size = p_to.size();
 
-			for (int j = 0; j < cf.methods.size(); j++) {
-				if (cf.methods[j].name != m.name) {
-					continue;
-				}
+	SearchArray<DocData::MethodDoc, MethodCmp> search_array;
 
-				const DocData::MethodDoc &mf = cf.methods[j];
+	for (const DocData::MethodDoc &from : p_from) {
+		int64_t found = search_array.bisect(to_ptrw, to_size, from, true);
 
-				m.description = mf.description;
-				m.is_deprecated = mf.is_deprecated;
-				m.deprecated_message = mf.deprecated_message;
-				m.is_experimental = mf.is_experimental;
-				m.experimental_message = mf.experimental_message;
-				m.keywords = mf.keywords;
-				break;
-			}
+		if (found >= to_size) {
+			continue;
 		}
 
-		for (int i = 0; i < c.signals.size(); i++) {
-			DocData::MethodDoc &m = c.signals.write[i];
+		DocData::MethodDoc &to = to_ptrw[found];
 
-			for (int j = 0; j < cf.signals.size(); j++) {
-				if (cf.signals[j].name != m.name) {
-					continue;
-				}
-				const DocData::MethodDoc &mf = cf.signals[j];
-
-				m.description = mf.description;
-				m.is_deprecated = mf.is_deprecated;
-				m.deprecated_message = mf.deprecated_message;
-				m.is_experimental = mf.is_experimental;
-				m.experimental_message = mf.experimental_message;
-				m.keywords = mf.keywords;
-				break;
-			}
+		// Check found entry on name.
+		if (to.name == from.name) {
```

Comment:
Is this check redundant with the searching above? Wouldn't the searcher only find a match if their names were equal?

Reply:
It's a binary search, so not guaranteed, it just ensures `from[found]` is *no greater* than `to`

Reply:
Ah, we might want to document that somewhere. It's not directly related to this issue, but I looked briefly at the code for `SearchArray` and it wasn't immediately apparent that would be the behavior in the case the query isn't found. I'm sure it's a standard thing for those who know, but for those who don't it's kind of surprising.

Reply:
That's 100% standard binary search though, that's a bog standard algorithm

Also it *is* documented [here](https://docs.godotengine.org/en/stable/classes/class_packedbytearray.html#class-packedbytearray-method-bsearch), and also [here](https://docs.godotengine.org/en/stable/classes/class_array.html#class-array-method-bsearch)

Reply:
Awesome :D

Reply:
Granted `SearchArray` is a bit more obscure, but it's very internal and the only reason I use it directly here is because `Vector::bsearch_custom` is non-const and does mutation checks :)

But might be worth documenting that class more directly, will see about that later

Reply:
Will add notes on those lines to `Vector::bsearch` and the `bisect` method after this :)